homeassistant/components/openrgb/light.py

```diff
@@ -0,0 +1,263 @@
+"""Platform for OpenRGB Integration."""
+import logging
+
+from openrgb import utils as RGBUtils
+
+# Import the device class from the component that you want to support
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_EFFECT,
+    ATTR_HS_COLOR,
+    DOMAIN as SENSOR_DOMAIN,
+    SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR,
+    SUPPORT_EFFECT,
+    LightEntity,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+import homeassistant.util.color as color_util
+
+from .const import (
+    DOMAIN,
+    ORGB_DISCOVERY_NEW,
+    SIGNAL_DELETE_ENTITY,
+    SIGNAL_UPDATE_ENTITY,
+)
+from .helpers import orgb_entity_id, orgb_icon, orgb_object_id, orgb_tuple
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up OpenRGB devices dynamically."""
+
+    async def async_discover_sensor(dev_ids):
+        """Discover and add a discovered openrgb sensor."""
+        if not dev_ids:
+            return
+
+        entities = await hass.async_add_executor_job(
+            _setup_entities,
+            hass,
+            dev_ids,
+        )
```

Comment:
I think _setup_entities is just doing data manipulation. Does it need to do this executor job jump to do that work elsewhere?